rp2040-hal/src/multicore.rs

```diff
@@ -290,3 +331,68 @@ impl<'p> Core<'p> {
         }
     }
 }
+
+#[cfg(all(target_arch = "arm", feature = "thread_local"))]
+mod thread_local {
+    use core::arch::global_asm;
+    use core::ptr::addr_of;
+
+    extern "C" {
+        static TLS_CORE_0: u8;
+        static TLS_CORE_1: u8;
+    }
+    // Not really a const pointer, but we reform it into mut in the asm
+    static mut TLS_STATE: [*const u8; 2] = [
+        // Point to linker-allocated space in .bss
+        unsafe { addr_of!(TLS_CORE_0) },
+        unsafe { addr_of!(TLS_CORE_1) },
+    ];
```

Comment:
Since we have a (small) set number of cores, I think we can do without a table and `_read_tp` to simply:
```
1. Load SIO::CPUID to r0
2. Compare with 0
3. Set `r0` to `addr_of!(TLS_CORE_0)` (preserves the flags set by 2.)
4. if equal flags set, skip next instr
5. Set `r0` to `addr_of!(TLS_CORE_1)`
6. return
```